### PR TITLE
Add PYTHONPATH modification logic for running python demos on Windows

### PR DESCRIPTION
**Pull request by nignatovsky**
Fixed incorrect PYTHONPATH modification with non-Windows delimiters on Windows.

### CODE REVIEW

File: demos/tests/run_tests.py [hunk 1]
```diff
@@ -149,11 +149,17 @@ def main():
         dl_dir = prepare_models(auto_tools_dir, args.downloader_cache_dir, args.mo, global_temp_dir, demos_to_test)
 
         num_failures = 0
+        
+        pythonpath_args = (os.environ['PYTHONPATH'], args.demo_build_dir)
+        
+        if os.name == "nt":
+            PYTHONPATH = {'PYTHONPATH': "{};{};".format(*pythonpath_args)}
+        else:
+            PYTHONPATH = {'PYTHONPATH': "{}:{}/lib".format(*pythonpath_args)}
```
Comment (alalek) [inline demos/tests/run_tests.py:158]: > `if os.name == "nt":`

Consider using `os.pathsep` for that instead.

Comment (nignatovsky) [reply demos/tests/run_tests.py]: Done.

File: demos/tests/run_tests.py [hunk 2]
```diff
@@ -149,11 +149,14 @@ def main():
         dl_dir = prepare_models(auto_tools_dir, args.downloader_cache_dir, args.mo, global_temp_dir, demos_to_test)
 
         num_failures = 0
+        
+        demo_build_subdir = "/lib" if os.pathsep == ":" else ""
+        pythonpath_args = (os.environ['PYTHONPATH'], os.pathsep, args.demo_build_dir, demo_build_subdir)
+        PYTHONPATH = {'PYTHONPATH': "{}{}{}{}".format(*pythonpath_args)}
```
Comment (IRDonch) [inline demos/tests/run_tests.py:155]: This variable isn't really necessary; you could just embed the value inside the `demo_environment` initializer.

Also, using an f-string would be more readable than `str.format`.

File: demos/tests/run_tests.py [hunk 3]
```diff
@@ -149,11 +149,14 @@ def main():
         dl_dir = prepare_models(auto_tools_dir, args.downloader_cache_dir, args.mo, global_temp_dir, demos_to_test)
 
         num_failures = 0
+        
+        demo_build_subdir = "/lib" if os.pathsep == ":" else ""
```
Comment (IRDonch) [inline demos/tests/run_tests.py:153]: The condition here seems rather indirect; we do not choose the path based on the path separator, we choose it based on the OS. So you should check the OS (you can use `platform.system()` to do that).

**Review by IRDonch — CHANGES_REQUESTED** [review]

File: demos/tests/run_tests.py (hunk 2 above)
Comment (nignatovsky) [reply demos/tests/run_tests.py]: Done.

File: demos/tests/run_tests.py (hunk 3 above)
Comment (nignatovsky) [reply demos/tests/run_tests.py]: Done.

File: demos/tests/run_tests.py [hunk 4]
```diff
@@ -149,11 +150,11 @@ def main():
...
 
         num_failures = 0
-
+        
+        demo_build_subdir = "" if platform.system() == "Windows" else "/lib"
         demo_environment = {**os.environ,
             'PYTHONIOENCODING': 'utf-8',
-            'PYTHONPATH': "{}:{}/lib".format(os.environ['PYTHONPATH'], args.demo_build_dir),
-        }
+            'PYTHONPATH': f"{os.environ['PYTHONPATH']}{os.pathsep}{args.demo_build_dir}{demo_build_subdir}"}
```
Comment (IRDonch) [inline demos/tests/run_tests.py:157]:
```suggestion
            'PYTHONPATH': f"{os.environ['PYTHONPATH']}{os.pathsep}{args.demo_build_dir}{demo_build_subdir}",
        }
```

File: demos/tests/run_tests.py [hunk 5]
```diff
@@ -149,11 +150,11 @@ def main():
         dl_dir = prepare_models(auto_tools_dir, args.downloader_cache_dir, args.mo, global_temp_dir, demos_to_test)
 
         num_failures = 0
-
+        
+        demo_build_subdir = "" if platform.system() == "Windows" else "/lib"
```
Comment (IRDonch) [inline demos/tests/run_tests.py:154]:
```suggestion
        python_module_subdir = "" if platform.system() == "Windows" else "/lib"
```

**Review by IRDonch — CHANGES_REQUESTED** [review]

**Review by IRDonch — APPROVED** [review]

**Comment by IRDonch** [comment]
Jenkins: test this please

**Pull request merged**